homeassistant/components/light/rest.py

```diff
@@ -0,0 +1,440 @@
+"""
+import asyncio
+import logging
+
+import ast
+import json
+
+import aiohttp
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, ATTR_COLOR_TEMP, SUPPORT_COLOR_TEMP,
+    ATTR_EFFECT, SUPPORT_EFFECT, ATTR_RGB_COLOR, SUPPORT_RGB_COLOR,
+    ATTR_TRANSITION, SUPPORT_TRANSITION, Light, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_NAME, CONF_URL_ON, CONF_URL_OFF, CONF_URL_STATE,
+    CONF_TIMEOUT, CONF_METHOD, CONF_USERNAME, CONF_PASSWORD)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.template import Template
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_BODY_OFF = 'body_off'
+CONF_BODY_ON = 'body_on'
+CONF_BODY_BRIGHTNESS = 'body_brightness'
+CONF_BODY_COLOR_TEMP = 'body_color_temp'
+CONF_BODY_EFFECT = 'body_effect'
+CONF_BODY_RGB_COLOR = 'body_rgb_color'
+CONF_BODY_TRANSITION = 'body_transition'
+CONF_IS_ON_TEMPLATE = 'is_on_template'
+CONF_BRIGHTNESS_TEMPLATE = 'brightness_template'
+CONF_COLOR_TEMP_TEMPLATE = 'color_temp_template'
+CONF_EFFECT_TEMPLATE = 'effect_template'
+CONF_RGB_COLOR_TEMPLATE = 'rgb_color_template'
+CONF_NAME_TEMPLATE = 'name_template'
+CONF_SUPPORTED_FEATURES_TEMPLATE = 'supported_features_template'
+CONF_SUPPORTED_FEATURES = 'supported_features'
+CONF_EFFECT_LIST_TEMPLATE = 'effect_list_template'
+
+DEFAULT_METHOD = 'post'
+DEFAULT_BODY_OFF = Template('{"is_on": false}')
+DEFAULT_BODY_ON = Template('{"is_on": true}')
+DEFAULT_BODY_BRIGHTNESS = Template('{"brightness": %d}')
+DEFAULT_BODY_COLOR_TEMP = Template('{"color_temp": %d}')
+DEFAULT_BODY_EFFECT = Template('{"effect": "%s"}')
+DEFAULT_BODY_RGB_COLOR = Template('{"rgb_color": [%d, %d, %d]}')
+DEFAULT_BODY_TRANSITION = Template('{"transition": %d}')
+DEFAULT_IS_ON_TEMPLATE = Template('{{value_json.is_on}}')
+DEFAULT_BRIGHTNESS_TEMPLATE = Template('{{value_json.brightness}}')
+DEFAULT_COLOR_TEMP_TEMPLATE = Template('{{value_json.color_temp}}')
+DEFAULT_EFFECT_TEMPLATE = Template('{{value_json.effect}}')
+DEFAULT_RGB_COLOR_TEMPLATE = Template('{{value_json.rgb_color}}')
+DEFAULT_NAME_TEMPLATE = Template('{{value_json.name}}')
+DEFAULT_SUPPORTED_FEATURES_TEMPLATE = Template(
+    '{{value_json.supported_features}}')
+DEFAULT_EFFECT_LIST_TEMPLATE = Template('{{value_json.effect_list}}')
+DEFAULT_SUPPORTED_FEATURES = ['brightness']
+DEFAULT_NAME = 'REST Light'
+DEFAULT_TIMEOUT = 10
+
+SUPPORT_REST_METHODS = ['post', 'put']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL_ON): cv.url,
+    vol.Required(CONF_URL_OFF): cv.url,
+    vol.Required(CONF_URL_STATE): cv.url,
+    vol.Optional(CONF_BODY_OFF, default=DEFAULT_BODY_OFF): cv.template,
+    vol.Optional(CONF_BODY_ON, default=DEFAULT_BODY_ON): cv.template,
+    vol.Optional(CONF_BODY_BRIGHTNESS,
+                 default=DEFAULT_BODY_BRIGHTNESS): cv.template,
```

Comment:
All these defaults don't work anymore. We have recently upgraded voluptuous and all defaults will be passed through the validator. So make all defaults just strings and they will be converted to template.